### PR TITLE
Add mx-puppet-instagram

### PR DESCRIPTION
**Pull request by jdreichmann**
Adds https://github.com/Sorunome/mx-puppet-instagram.

Still needs documentation and testing

### CODE REVIEW

File: group_vars/matrix_servers [hunk 1]
```diff
@@ -392,6 +392,37 @@ matrix_mx_puppet_slack_login_shared_secret: "{{ matrix_synapse_ext_password_prov
...
+matrix_mx_puppet_instagram_container_image_self_build: "{{ matrix_architecture != 'amd64'}}"
+
+matrix_mx_puppet_instagram_systemd_required_services_list: |
+  {{
+    ['docker.service']
+    +
+    (['matrix-synapse.service'] if matrix_synapse_enabled else [])
+  }}
+
+matrix_mx_puppet_instagram_appservice_token: "{{ matrix_synapse_macaroon_secret_key | password_hash('sha512', 'mxslk.as.tok') | to_uuid }}"
```
Comment (spantaleev) [inline group_vars/matrix_servers:413]: The `mxslk.as.tok` key here and `mxslk.hs.tok` in `matrix_mx_puppet_instagram_homeserver_token` conflict with the keys used for the Slack bridge. If both are enabled, Synapse will complain that tokens are not unique.

Let's use a different key for this bridge. It should be a relatively short string to be compatible with various versions of Ansible (e.g. `insta.as.tok` and `insta.hs.tok`).

Comment (jdreichmann) [reply group_vars/matrix_servers]: Adressed in 2073cd6

**Comment by jdreichmann** [comment]
Added some documentation and thanks @spantaleev for the hint with the colliding access-tokens, i overlooked that

**Comment by spantaleev** [comment]
Looks good! Think we're ready to merge this or would you like to test it some more first?

**Comment by jdreichmann** [comment]
I tested it on my HS and then marked it ready for review so other can test if they want, but from my side i consider it ready for merge.

**Comment by spantaleev** [comment]
That's good enough! Thanks a lot for increasing the number of bridges we support! :+1:

**Pull request merged**